test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -146,6 +146,37 @@ call_log_manager::process_call_log_entry(json call_log_entry)
             }
             break;
         }
+        case api_method::begin_transaction: {
+            const std::string session_id = call_log_entry["session_id"].get<std::string>();
+            /*
+             * Not having a valid connection is a fatal error since no other operations can happen
+             * without a connection.
+             */
+            if (_conn == nullptr)
```

Comment:
Be careful with the other PR in parallel regarding this check

Reply:
Good point, I'll wait for the other PR to merge first and remove that check here.